File: clients/Java-GraalVM-native/src/main/java/model/Player.java
package model;

import util.StreamUtil;

public class Player {
    private int id;
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    private int score;
    public int getScore() { return score; }
    public void setScore(int score) { this.score = score; }
    private int resource;
    public int getResource() { return resource; }
    public void setResource(int resource) { this.resource = resource; }
    public Player() {}
    public Player(int id, int score, int resource) {
        this.id = id;
        this.score = score;
        this.resource = resource;
    }
    public static Player readFrom(java.io.InputStream stream) throws java.io.IOException {
        Player result = new Player();
        result.id = StreamUtil.readInt(stream);
        result.score = StreamUtil.readInt(stream);
        result.resource = StreamUtil.readInt(stream);
        return result;
    }
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeInt(stream, id);
        StreamUtil.writeInt(stream, score);
        StreamUtil.writeInt(stream, resource);
    }
}

File: clients/Kotlin-GraalVM-native/src/main/kotlin/model/Player.kt
package model

import util.StreamUtil

class Player {
    var id: Int = 0
    var score: Int = 0
    var resource: Int = 0
    constructor() {}
    constructor(id: Int, score: Int, resource: Int) {
        this.id = id
        this.score = score
        this.resource = resource
    }
    companion object {
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Player {
            val result = Player()
            result.id = StreamUtil.readInt(stream)
            result.score = StreamUtil.readInt(stream)
            result.resource = StreamUtil.readInt(stream)
            return result
        }
    }
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, id)
        StreamUtil.writeInt(stream, score)
        StreamUtil.writeInt(stream, resource)
    }
}
